build_from.py

```diff
@@ -72,11 +78,53 @@ def generate_diffs(reg_tree, act_title_and_section, builder, layer_cache):
         for rhs_version, rhs_tree in all_versions.iteritems():
             changes = checkpointer.checkpoint(
                 "-".join(["diff", lhs_version, rhs_version]),
-                lambda: dict(changes_between(lhs_tree, rhs_tree)))
+            lambda: dict(changes_between(lhs_tree, rhs_tree)))
             writer.diff(
                 label_id, lhs_version, rhs_version
             ).write(changes)
 
+def build_by_notice(filename, title, doc_number, act_title, act_section, notice_doc_numbers, checkpoint=None):
```

Comment:
Yes, in general, I should refactor all this into a single sensible driver function. I've just been too lazy to do so.